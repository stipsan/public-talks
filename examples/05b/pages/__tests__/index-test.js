import renderer from 'react-test-renderer'
import snapshotDiff from 'snapshot-diff'
import App from '../index'

it('should open and close', () => {
  const component = renderer.create(<App />)
  expect(component.toJSON()).toMatchSnapshot()

  component.root.findByType('button').props.onClick()
  expect(component.toJSON()).toMatchSnapshot()

  expect(snapshotDiff(<App />, <App isOpen />)).toMatchSnapshot()
})
