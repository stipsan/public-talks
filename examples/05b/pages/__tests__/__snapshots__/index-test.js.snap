// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should open and close 1`] = `
<div>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`should open and close 2`] = `
<div>
  <div>
    Modal content
  </div>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`should open and close 3`] = `
Snapshot Diff:
- <Modal isOpen={false} />
+ <Modal isOpen={true} />

  <div>
+   <div>
+     Modal content
+   </div>
    <button
      onClick={[Function onClick]}
    />
  </div>
`;
